tests/Microsoft.Identity.Test.Unit/BrokerTests/BrokerRequestTests.cs

```diff
@@ -378,28 +378,68 @@ public void NoTokenFoundThrowsUIRequiredTest()
         {
             using (var harness = CreateBrokerHelper())
             {
+                try
+                {
+                    _brokerSilentAuthStrategy.ValidateResponseFromBroker(CreateErrorResponse(BrokerResponseConst.NoTokenFound));
+                }
+                catch (MsalUiRequiredException ex)
+                {
+                    Assert.IsTrue(ex.ErrorCode == BrokerResponseConst.NoTokenFound);
+                    return;
+                }
 
-                var response = new MsalTokenResponse
+                Assert.Fail("Wrong Exception thrown. ");
+            }
+        }
+
+        [TestMethod]
+        public void NoAccountFoundThrowsUIRequiredTest()
```

Comment:
I was thinking that the best place to add this the logic you've added is actually in AndroidBroker.cs; However we can't test that. Having tests beats a marginally better architecture.

Reply:
+1 to the unit testing, especially since not sure how to manually test w/out messing up my authenticator app.

---
In reply to: [526108961](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2200#discussion_r526108961) [](ancestors = 526108961)